internal/extensions/actions/call_files_to_save.go

```diff
@@ -0,0 +1,141 @@
+package actions
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/kyma-project/cli.v3/internal/clierror"
+	"github.com/kyma-project/cli.v3/internal/cmdcommon"
+	"github.com/kyma-project/cli.v3/internal/extensions/actions/common"
+	"github.com/kyma-project/cli.v3/internal/extensions/call"
+	"github.com/kyma-project/cli.v3/internal/extensions/types"
+	"github.com/spf13/cobra"
+)
+
+type targetPodConfig struct {
+	Selector  map[string]string `yaml:"selector"`
+	Namespace string            `yaml:"namespace"`
+	Port      string            `yaml:"port"`
+	Path      string            `yaml:"path"`
+}
+
+type requestConfig struct {
+	// TODO: support other HTTP methods
+	// Method    string            `yaml:"method"`
+	Parameters map[string]string `yaml:"parameters"`
+}
+
+type callFilesToSaveConfig struct {
+	Request   requestConfig   `yaml:"request"`
+	TargetPod targetPodConfig `yaml:"targetPod"`
+	OutputDir string          `yaml:"outputDir"`
+}
+
+func (c *callFilesToSaveConfig) validate() clierror.Error {
+	if c.TargetPod.Namespace == "" {
+		return clierror.New("empty target pod namespace")
+	}
+	if c.TargetPod.Selector == nil {
+		return clierror.New("empty target pod selector")
+	}
+	if c.TargetPod.Port == "" {
+		return clierror.New("empty target pod port")
+	}
+	if c.TargetPod.Path == "" {
+		return clierror.New("empty target pod path")
+	}
+	if c.OutputDir == "" {
+		return clierror.New("empty output directory path")
+	}
+
+	return nil
+}
+
+type callFilesToSaveAction struct {
+	common.TemplateConfigurator[callFilesToSaveConfig]
+
+	kymaConfig *cmdcommon.KymaConfig
+}
+
+func NewCallFilesToSaveAction(kymaConfig *cmdcommon.KymaConfig) types.Action {
+	return &callFilesToSaveAction{
+		kymaConfig: kymaConfig,
+	}
+}
+
+func (a *callFilesToSaveAction) Run(cmd *cobra.Command, _ []string) clierror.Error {
+	clierr := a.Cfg.validate()
+	if clierr != nil {
+		return clierror.WrapE(clierr, clierror.New("invalid action configuration"))
+	}
+
+	if !filepath.IsLocal(a.Cfg.OutputDir) {
+		// output dir is not a local path, ask user for confirmation
+		clierr = getUserAcceptance(a.Cfg.OutputDir)
```

Comment:
`getUserAcceptance` is defined in the `function_init.go` - if user won't accept the prompt then the returned value will be `function init aborted` which might be misleading